interface/forms/vitals/templates/vitals/general_new.html

```diff
@@ -121,213 +134,239 @@
-                {if $units_of_measurement == 1}<td class="valuesunfocus">{else}<td class='currentvalues'>{/if}
-                <input type="text" size='5' id='waist_circ_input_metric' value="{if $vitals->get_waist_circ() != 0}{math equation="round(number * constant,1)" number=$vitals->get_waist_circ() constant=2.54 format="%.2f"}{/if}" onChange="convCmtoIn('waist_circ_input');"/>
-                </td>
-        {foreach item=result from=$results}
-                <td class='historicalvalues'>{if $result.waist_circ != 0}{math equation="number * constant" number=$result.waist_circ constant=2.54 format="%.2f"}{/if}</td>
-        {/foreach}</tr>
-
-	<tr><td class="graph" id="BMI">{xl t="BMI"}</td><td>{xl t="kg/m^2"}</td>
-		<td class='currentvalues'><input type="text" size='5'
-			name='BMI' id='BMI_input' value="{if $vitals->get_BMI() != 0}{$vitals->get_BMI()|substr:0:5}{/if}"/></td>
-	{foreach item=result from=$results}
-		<td class='historicalvalues'>{if $result.BMI != 0}{$result.BMI|substr:0:5}{/if}</td>
-	{/foreach}</tr>
-
-  <tr><td>{xl t="BMI Status"}</td><td>{xl t="Type"}</td>
-    <td class='currentvalues'><input type="text" size='15'
-    name="BMI_status" id="BMI_status' value="{$vitals->get_BMI_status()}"/></td>
-  {foreach item=result from=$results}
-    <td  class='historicalvalues'>{if $result.BMI_status}{xl t=$result.BMI_status}{/if}</td>
-  {/foreach}
-  </tr>
-
-  <tr><td>{xl t="Other Notes"}<td></td></td>
-    <td class='currentvalues'><input type="text" size='20'
-      name="note" id='note' value="{$vitals->get_note()}" /></td>
-  {foreach item=result from=$results}
-    <td class='historicalvalues'>{$result.note}</td>
-  {/foreach}</tr>
-
-	<tr>
-	<td colspan='3' style='text-align:center'>
-	{if $patient_age <= 20 || (preg_match('/month/', $patient_age))}
-	<!-- only show growth-chart button for patients < 20 years old -->
-	<!-- <input type="button" id="growthchart" value="{xl t="Growth-Chart"}" style='margin-left: 20px;'> -->
-	<input type="button" id="pdfchart" value="{xl t="Growth-Chart"} ({xl t="PDF"})" style='margin-left: 20px;'>
-        <input type="button" id="htmlchart" value="{xl t="Growth-Chart"} ({xl t="HTML"})" style='margin-left: 20px;'>
-	{/if}
-	</td>
-	</tr>
-
-	<tr><td colspan='3' style='text-align:center'>&nbsp;</td></tr>
-	<tr>
-	<td colspan='3' style='text-align:center'>
-        <input type="submit" class="editonly" name="Submit" value="{xl t="Save"}">
-        <input type="button" class="editonly" id="cancel" value="{xl t="Cancel"}">
-        </td>
-</tr>
-</table>
-<br><br>
-<input type="hidden" name="id" id='id' value="{$vitals->get_id()}" />
-<input type="hidden" name="activity" id='activity' value="{$vitals->get_activity()}">
-<input type="hidden" name="pid" id='pid' value="{$vitals->get_pid()}">
-<input type="hidden" name="process" id='process' value="true">
-</form>
+
+ <div class="container">
+        <div class="row">
+            <div class="col-sm-12">
```

Comment:
We should try avoiding size qualifiers in bootstrap unless we are going to implement viewport specific layouts.  When you specify col-sm*, it translates into a fixed width for small devices.  That cause wasted space for md and lg.  

As a standard, it is better to stick to col-12 rather than col-md-12 or col-sm-12.  That will let the library dynamically adjust breaks on different form factors or browser window sizes.

Reply:
Is ````col-12```` a BS3 class or only a BS4 class?

Reply:
Keep forgetting in bs4, design for xs form factor became standard as seen in these two bookmarks :  [bs3 grid options](https://bootstrapdocs.com/v3.3.6/docs/css/#grid-options) vs [bs4 grid options](https://getbootstrap.com/docs/4.0/layout/grid/#grid-options)

Since the classes get applied in order, it may be worth trying the "col-xs-4 col-4" combination to work in bs4 world without having to revisit every script.  As they note, if you choose **sm** classes, in bs4 it would have effect of **md** in bs3.

Quote from their migration document: 
xs grid classes have been modified to not require the infix to more accurately represent that they start applying styles at min-width: 0 and not a set pixel value. Instead of .col-xs-6, it’s now .col-6. All other grid tiers require the infix (e.g., sm).